text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,277 @@
+| Enclosed in parentheses.                                           | US                 | ($127.54)  |
+
+Most currencies use the same decimal and thousands separator that other numbers in the locale use, but this is not always true. In some places in Switzerland, they use the period as a decimal separator for Swiss francs (Sfr. 127.54), but then use commas as the decimal separator everywhere else (127,54).
+
+We take the point of view that these differences are not "token metadata", which are determined by the ID of the token that we are using, but rather localization properties determined by the locale of the user. So, for example the same app might be capable of displaying `kr 127,54` and `127.54 kr`, depending on the locale of the user, and token metadata does not influence this.
+
+This is also consistent with [how shopify handles currency localization](https://polaris.shopify.com/foundations/formatting-localized-currency).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The token-metadata json fields have the following semantics:
+
+* `schema_version`: An integer describing the version number of the schema of this document. This is 1 until a future MCIP proposes schema modifications.
+* `signing_timestamp`: A string representation of the unix timestamp at the (approximate) time of creating the document's signature. This is an integer number of seconds since the unix epoch.
+* `token_id`: A decimal string representation of the `u64` corresponding to the token id.
+* `currency_name`: A UTF-8 string which is the full, official name of the currency. This is intended to be an English language noun or noun phrase.
+* `short_code`: A short code for the currency. This may be displayed to users in a manner similar to ISO 4217 currency codes, for example, USD, GBP, CAD, in connection to an amount of currency. This is expected to match the ticker symbol used for this asset on cryptocurrency exchanges. Not more than 12 printable ASCII characters, including `[A-Za-z0-9].-`.
+* `symbol`: Most fiat currencies have a printable character such as $, £, ¥. Some cryptocurrencies do also. Bitcoin has ₿. Ethereum has Ξ. This may be optionally specified as a UTF-8 string in the "symbol" field.
+* `decimals`: An integer specifying how the `u64` integer in a TxOut in the blockchain is scaled to compute a user-displayed amount of the currency. For example, MOB has 12 decimals, which indicates that `10^12` of the smallest representable units of MOB on the MobileCoin blockchain are equal to one MOB.
+* `logo_svg`: An optional logo image. This is a base64-encoded SVG document. This is expected to have been sanitized using something like [svg-hush](https://github.com/cloudflare/svg-hush), to remove scripting, hyperlinks to other documents, and references to cross-origin resources. The full extent of such sanitization will not be specified here.
```

Comment:
On further thought, how about we omit the base64? That would make it easy for humans to verify the SVG has safe elements, as well as being 25% smaller and gzipping better: https://css-tricks.com/probably-dont-base64-svg/

Reply:
Honestly i feel like we should not do inline SVG, we should just have a url for the SVG and a hash of the SVG data. I think base64 and not base64 are both pretty human (and internet) unfriendly

Reply:
Fine by me, as long as we include a hash/signature of the actual image.

Reply:
We may want to consider that a client won't always have internet connection, so if the images aren't cached, they won't be available.